theme/models.py

```diff
@@ -324,4 +325,36 @@ def force_unique_emails(sender, instance, **kwargs):
         if sender.objects.filter(username=username).exclude(pk=instance.id).exists():
             raise ValidationError("Username already in use.")
 
+
 pre_save.connect(force_unique_emails, sender=User)
+
+
+@receiver(models.signals.pre_save, sender=UserProfile)
+def auto_delete_file_on_change(sender, instance, **kwargs):
+    """
+    Deletes old file from filesystem when corresponding UserProfile object is updated with new file.
+    """
+    if not instance.pk:
+        return
+
+    try:
+        up = UserProfile.objects.get(pk=instance.pk)
+        if up:
+            old_file_cv = up.cv
+            old_file_pic = up.picture
+            if not old_file_cv and not old_file_pic:
+                return
+        else:
+            return
+    except UserProfile.DoesNotExist:
+        return
```

Comment:
This is a pretty big deal if we get here, I'd like it logged as a warning.  I don't think this will ever happen because of the instance.pk check earlier.